accounts-on-demand/src/subscription_manager.rs

```diff
@@ -0,0 +1,281 @@
+                                .pubkey
+                                .try_into()
+                                .expect("Pubkey should be 32 byte long");
+                            let owner: [u8; 32] = account_data
+                                .owner
+                                .try_into()
+                                .expect("owner pubkey should be deserializable");
+                            let notification = AccountNotificationMessage {
+                                data: AccountData {
+                                    pubkey: Pubkey::new_from_array(account_pk_bytes),
+                                    account: Arc::new(Account {
+                                        lamports: account_data.lamports,
+                                        data: account_data.data,
+                                        owner: Pubkey::new_from_array(owner),
+                                        executable: account_data.executable,
+                                        rent_epoch: account_data.rent_epoch,
+                                    }),
+                                    updated_slot: account.slot,
+                                },
+                                // TODO update with processed commitment / check above
+                                commitment: Commitment::Processed,
+                            };
+                            if account_stream_sx.send(notification).is_err() {
+                                // non recoverable, i.e the whole stream is being restarted
+                                log::error!("Account stream broken, breaking from main loop");
+                                break 'main_loop;
+                            }
+                        }
+                    }
+                    UpdateOneof::Ping(_) => {
+                        log::trace!("GRPC Ping accounts stream");
+                    }
+                    _ => {
+                        log::error!("GRPC accounts steam misconfigured");
+                    }
+                };
+
+                match restart_channel.try_recv() {
+                    Ok(_) => {
+                        log::info!("Restarting account subscription");
+                        break;
+                    }
+                    Err(e) => match e {
+                        broadcast::error::TryRecvError::Empty => {}
+                        broadcast::error::TryRecvError::Closed => {}
+                        broadcast::error::TryRecvError::Lagged(_) => {
+                            // remove message in the queue
+                            let _ = restart_channel.try_recv();
+                            break;
+                        }
+                    },
+                }
+            }
+        }
+        Ok(())
+    })
+}
+
+pub fn create_grpc_account_streaming(
+    grpc_sources: Vec<GrpcSourceConfig>,
+    accounts_filters: Arc<RwLock<AccountFilters>>,
```

Comment:
I would rather send the new filters via the **restart_channel**

Reply:
I would do microbatching (i.e. wait 500ms or so) and then consume and merge all requested accounts instead of trigger re-subscription per new filter list

Reply:
On multiple updates would always want last updated